### PR TITLE
Enable pubternal analyzer

### PR DESCRIPTION
**Pull request by pakrym**

### CODE REVIEW

**Comment by pakrym** [comment]
I was going to, deciding what to do with APIs is a separate conversation that might last a long time. I can file an issue but wouldn't like to block PR on it.

**Comment by Tratcher** [comment]
Then what's the benefit of this PR? To discourage any further exposure?

**Comment by pakrym** [comment]
Yes, for example we accidentally exposed https://github.com/aspnet/KestrelHttpServer/pull/2560/files#diff-4c4e31b7f64ff3373186c05d2ffa1ea1R46 in 2.1

File: src/Kestrel.Core/ListenOptions.cs [hunk 1]
```diff
@@ -17,7 +17,9 @@ namespace Microsoft.AspNetCore.Server.Kestrel.Core
     /// Describes either an <see cref="IPEndPoint"/>, Unix domain socket path, or a file descriptor for an already open
     /// socket that Kestrel should bind to or open.
     /// </summary>
+#pragma warning disable PUB0001 // Pubternal type in public API
     public class ListenOptions : IEndPointInformation, IConnectionBuilder
+#pragma warning restore PUB0001 // Pubternal type in public API
```
Comment (halter73) [inline src/Kestrel.Core/ListenOptions.cs]: It should be allowed to implement a pubternal interface like you can with an internal one.

**Comment by halter73** [comment]
No one merge until https://github.com/aspnet/KestrelHttpServer/pull/2543 is in dev.

**Comment by pakrym** [comment]
@halter73 what should I do with this PR now?

**Pull request merged**